src/dir_status.mli

```diff
@@ -1,5 +1,10 @@
 open Stdune
 
+type is_component_of_a_group_but_not_the_root = {
+  group_root : Path.t;
```

Comment:
One thing that is slightly weird to me is that you're returning a build path here, but every other constructor is return a source path for the root (`File_tree.Dir.t`). Is there a way to keep things consistent, perhaps? This way we'd be able to use this record as the payload for everyone of the constructors.

Reply:
I'm returning the same kind of path as what the caller gave me, which is I think what counts here. (at least that's what's necessary at the call site)

> we'd be able to use this record as the payload for everyone of the constructors

I'm not sure that's a good idea: I think `File_tree.Dir.t` gives you much more than just the path and if the call sites don't need it we probably shouldn't give it to them.